### PR TITLE
feat: change default `OutputsValidator` to `well_known_scripts_only`

### PR DESCRIPTION
**Pull request by quake**

### CODE REVIEW

File: spec/src/consensus.rs [hunk 1]
```diff
@@ -536,6 +550,8 @@ pub struct Consensus {
     pub permanent_difficulty_in_dummy: bool,
     /// A switch to select hard fork features base on the epoch number.
     pub hardfork_switch: HardForkSwitch,
+    pub well_known_lock_scripts: Vec<Script>,
+    pub well_known_type_scripts: Vec<Script>,
```
Comment (keroro520) [inline spec/src/consensus.rs:554]: According to my understanding, `Consensus` serves as an identifier of a blockchain. I don't understand why `well_known-*_ scripts` should be included in `Consensus`?

Comment (yangby-cryptape) [reply spec/src/consensus.rs:554]: I totally agree with @keroro520.

It's not a part of `Consensus`. It belongs to RPC module.

Comment (quake) [reply spec/src/consensus.rs]: moved the related code to `PoolRpcImpl`, please review again, thanks.

File: util/jsonrpc-types/src/pool.rs [hunk 2]
```diff
@@ -70,13 +70,13 @@ impl From<CorePoolTransactionEntry> for PoolTransactionEntry {
...
     ///
-    /// The default validator only allows outputs (a.k.a., cells) that
+    /// This validator only allows outputs (a.k.a., cells) that
     ///
-    /// * use either the secp256k1 or the secp256k1 multisig bundled in the genesis block via type script hash as the lock script,
-    /// * and the type script is either empty or DAO.
-    Default,
+    /// * use `code_hash` and `hash_type` listed in https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0024-ckb-system-script-list/0024-ckb-system-script-list.md as the lock or type script,
+    /// * use `code_hash` and `hash_type` listed in `custom_well_known_scripts` of `ckb.toml`
+    WellKnownScriptsOnly,
```
Comment (keroro520) [inline util/jsonrpc-types/src/pool.rs:79]: [The generated README](https://github.com/nervosnetwork/ckb/pull/2821/files#diff-6fe1f4e31ab0e32a0eb34735313912f3bf54073b53bb7d67b9af1f41d3f89c44R3788-R3790) does not include this part. Is it a bug?

File: rpc/src/module/pool.rs [hunk 3]
```diff
@@ -241,17 +331,22 @@ impl PoolRpc for PoolRpcImpl {
...
-            Some(OutputsValidator::Passthrough) | None => Ok(()),
+            Some(OutputsValidator::Passthrough) => Ok(()),
         } {
             return Err(RPCError::custom_with_data(
                 RPCError::PoolRejectedTransactionByOutputsValidator,
                 format!(
                     "The transction is rejected by OutputsValidator set in params[1]: {}. \
-                    Please set it to passthrough if you really want to send transactions with advanced scripts.",
-                    outputs_validator.unwrap_or(OutputsValidator::Default).json_display()
+                    Please check the related information in https://github.com/nervosnetwork/ckb/wiki/Transaction-%C2%BB-Default-Outputs-Validator",
```
Comment (keroro520) [inline rpc/src/module/pool.rs:348]: * Wiki should add description of `extra_well_known_type_scripts` and `extra_well_known_lock_scripts`

* Tell users which script is errored(most developers of DApps will encounter this error at least once.)

Comment (quake) [reply rpc/src/module/pool.rs:348]: I plan to modify this wiki page after this PR and [cheque] rfc PR (https://github.com/nervosnetwork/rfcs/pull/241) are merged.

Comment (quake) [reply rpc/src/module/pool.rs:348]: > * Tell users which script is errored(most developers of DApps will encounter this error at least once.)

RPC will report the error output index in line 475
```rust
.map_err(|err| format!("output index: {}, error: {:?}", index, err))
```

File: util/jsonrpc-types/src/pool.rs (hunk 2 above)
Comment (quake) [reply util/jsonrpc-types/src/pool.rs:79]: it looks like an existing bug of `rpc.py` which cannot generate multiple line text with enum variant, for example, compare the doc of `DepType`:
https://github.com/nervosnetwork/ckb/blob/develop/rpc/README.md#type-deptype
https://github.com/nervosnetwork/ckb/blob/develop/util/jsonrpc-types/src/blockchain.rs#L276-L278

let me try to fix it in a new PR and re-generate the readme file for all variants.

File: rpc/src/module/pool.rs (hunk 3 above)
Comment (keroro520) [reply rpc/src/module/pool.rs:348]: An output has one lock-script and one type-script. Is it possible to distinguish between lock-script and type-script errors?

Comment (quake) [reply rpc/src/module/pool.rs:348]: resolved in commit https://github.com/nervosnetwork/ckb/pull/2821/commits/deedc21f03c0e0737e0dc754df8a95ce351b7d91

**Review by keroro520 — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Review by zhangsoledad — APPROVED** [review]

**Comment by quake** [comment]
bors r=doitian,keroro520,zhangsoledad

**Pull request merged**